tasks/managed/add-fbc-contribution/add-fbc-contribution.yaml

```diff
@@ -297,21 +305,236 @@ spec:
+                if [ "$memory_limit_bytes" -eq 0 ] || [ "$memory_limit_bytes" -gt 107374182400 ]; then
+                    echo "WARN: No memory limit detected or limit too high, using conservative default" >&2
+                    echo "3"
+                    return
+                fi
+                
+                # Convert to MiB
+                memory_limit_mib=$((memory_limit_bytes / 1024 / 1024))
+                echo "INFO: Memory limit: ${memory_limit_mib}Mi" >&2
+            fi
+            
+            # Estimated memory usage per worker (based on observed usage):
+            # - JSON processing with jq: ~30-40Mi
+            # - kubectl operations and output buffers: ~30-40Mi
+            # - Bash subprocess overhead: ~10-20Mi
+            # - Safety margin: 20%
+            # Total: ~100Mi per worker conservatively
+            local memory_per_worker_mib=100
+            
+            # Reserve base memory for main script operations
+            # - Initial snapshot loading and parsing: ~50Mi
+            # - Results aggregation: ~30Mi
+            # - System overhead: ~50Mi
+            local base_memory_mib=130
+            
+            local available_for_workers=$((memory_limit_mib - base_memory_mib))
+            
+            if [ "$available_for_workers" -lt "$memory_per_worker_mib" ]; then
+                echo "WARN: Very low memory available (${available_for_workers}Mi), using single worker" >&2
+                echo "1"
+                return
+            fi
+            
+            local optimal_workers=$((available_for_workers / memory_per_worker_mib))
+            
+            local max_cap=8
+            if [ "$optimal_workers" -gt "$max_cap" ]; then
+                echo "INFO: Calculated $optimal_workers workers, capping at $max_cap to avoid IIB overload" >&2
+                optimal_workers=$max_cap
+            fi
+            
+            if [ "$optimal_workers" -lt 1 ]; then
+                optimal_workers=1
+            fi
+            
+            echo "INFO: Adaptive parallelism: ${memory_limit_mib}Mi limit, ${memory_per_worker_mib}Mi per worker" >&2
+            echo "INFO: Calculated optimal parallel workers: $optimal_workers" >&2
+            
+            echo "$optimal_workers"
+        }
 
-            for ocp_version in $ocp_versions; do
+        # Process all OCP groups in parallel with bounded parallelism
+        process_all_ocp_groups() {
+            echo "INFO: Starting parallel group processing with adaptive worker limits"
+
+            local parallel_dir
+            parallel_dir="$(params.dataDir)/parallel-tracking"
+            mkdir -p "$parallel_dir"
+            
+            export parallel_dir
+
+            # Calculate optimal parallelism based on available memory
+            local max_parallel_workers
+            max_parallel_workers=$(calculate_max_parallel_workers)
+            
+            # shellcheck disable=SC2206
```

Comment:
The disables need to be more explicit about why you are disabling it. The code says it is SC2206, but the better is having an actual description about it. You can search the catalog to see how we approach this (at least afaik).